### ISSUE TITLE
Support multiple aggregations

### ISSUE THREAD

**Issue by lukaspawlik** [10up/ElasticPress#636]
It would be nice to add a support for multiple aggregations.

**Comment by charlesLF** [comment]
The way I've been doing multiple aggregations was with something like this:

```
$aggs = array(
    'name' => "my_aggregation",
    'use-filter' => false,
    'aggs' => array(
        "subagg_1" => array(
                    'terms' => array(
                        'field' => "field1.name"
                        )
                    ),
        "subagg_2" => array(
                    'terms' => array(
                        'field' => "field2.name"
                        )
                    ),            
        )
    );

$query->query_vars["aggs"] = $aggs;
```

How would the syntax become with this pull request?

**Comment by Ritesh-patel** [comment]
Hi @charlesLF 

Syntax for that PR will be as following:

```
$aggs = array(
    array(
        'name' => 'subagg_1',
        'use-filter' => false,
        'aggs' => array(
            'terms' => array(
                'size' => 100,
                'field' => "field1.name",
            ),
        )
    ),
    array(
        'name' => 'subagg_2',
        'aggs' => array(
            'terms' => array(
                'size' => 100,
                'field' => "field2.name",
            ),
        )
    ),
),

$query->query_vars["aggs"] = $aggs;
```

**Comment by lukaspawlik** [comment]
@charlesLF could you tell me what Elasticsearch version do you use? I wasn't able to make your query aggregations arguments working.

**Comment by charlesLF** [comment]
Hey @lukaspawlik,

I just saw your comment. I am using Elasticsearch 2.3.4

This is something that I have done a while ago, so I had forgotten about all the workarounds that were involved.

I actually had to use an extra function to "fix" the aggregation buckets, when "use-filter" was false:

```
add_filter( 'ep_formatted_args', array('add_search_aggs'), 10, 1 );

function add_search_aggs($formatted_args) {
        // This is a (temporary) workaround for using multiple aggregations. 
        // First, make sure this filter is only being applied on my_aggregation, and only for the case where use-filter is set to false
        if(isset($formatted_args["aggs"]["my_aggregation"]["use-filter"]) && $formatted_args["aggs"]["my_aggregation"]["use-filter"] == false) {

            // these are longer be needed after #634.
            unset($formatted_args["aggs"]["my_aggregation"]["name"]);
            unset($formatted_args["aggs"]["my_aggregation"]["use-filter"]);

           // Elasticsearch wants a "filter" to be set, even if it's just an empty object
           // Also, the sub aggregations should be wrapped with an extra "aggs"    
            $object = new stdClass();
            $formatted_args["aggs"]["my_aggregation"] = array(
                       "filter" => $object,
                        "aggs" => $formatted_args["aggs"]["my_aggregation"]["aggs"]
                    );
        }

        return $formatted_args;
        
    }
```

Note: The above function would not be needed if "use-filter" was true.

To explain, the function removes the "name" & "use-filter" attributes from $formatted_args (this is no longer needed since it's fixed in #634), and then wraps the buckets with an "aggs" so that the final formatted arguments for "aggs" is something like:

```
$formatted_args = array(
...
"aggs" => array(
        "my_aggregation" => array (
            "aggs" => array(
                "subagg_1" => array("terms" => array("field" => "field_name1.raw")),
                "subagg_2" => array("terms" => array("field" => "field_name2.raw")),
                ...
                )
            )
    )
...
);
```

And that's how I was able to get multiple aggregations working :smile:

**Comment by charlesLF** [comment]
Long story short, since #634 fixes the first issue related to "name" and "use-filter" incorrectly appearing in `$formatted_args`, then the only thing that is still required for multiple aggregations to work with the syntax I mentioned in my earlier comment is by adding an empty filter to the aggregation when `use-filter` is false:

```
$formatted_args["aggs"]["my_aggregation"]['filter'] = new stdClass();
```

**Comment by charlesLF** [comment]
Here's the additional syntax if you have #634 applied:

```
add_filter( 'ep_formatted_args', 'add_search_aggs', 10, 1 ); 

public function add_search_aggs($formatted_args) {
        if(!isset($formatted_args["aggs"]["my_aggregation"]['filter'])) {
            
            $formatted_args["aggs"]["my_aggregation"] = array(
                'filter' => new stdClass(),
                'aggs' => $formatted_args["aggs"]["my_aggregation"]
                );
        }
        return $formatted_args;
        
    }
```
